pkg/monitor/cluster/nodes.go

```diff
@@ -0,0 +1,57 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type metricsLabel string
+
+const (
+	notReadyNodes           metricsLabel = "notReadyNodes"
+	memoryPressureNodes     metricsLabel = "memoryPressureNodes"
+	diskPressureNodes       metricsLabel = "diskPressureNodes"
+	pidPressureNodes        metricsLabel = "pidPressureNodes"
+	networkUnavailableNodes metricsLabel = "networkUnavailableNodes"
+)
+
+var metricsLabelMap = map[corev1.NodeConditionType]metricsLabel{
+	corev1.NodeReady:              notReadyNodes,
+	corev1.NodeMemoryPressure:     memoryPressureNodes,
+	corev1.NodeDiskPressure:       diskPressureNodes,
+	corev1.NodePIDPressure:        pidPressureNodes,
+	corev1.NodeNetworkUnavailable: networkUnavailableNodes,
+}
+
+func (mon *Monitor) emitNodesMetrics() error {
+	nodes, err := mon.cli.CoreV1().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	mon.emitGauge("cluster.nodes.total", int64(len(nodes.Items)), nil)
+
+	counters := map[metricsLabel]int64{}
+	for _, node := range nodes.Items {
+		for _, c := range node.Status.Conditions {
+			// count 'Unknown' status as unhealthy state for each condition. In this way
+			// we can flag issues without creating additional timeseries for each condition.
+			// for NodeReady count a node when the status is False (not ready) or Unknown
+			// for other conditions count when the status is True or Unknown
+			if c.Type == corev1.NodeReady && c.Status != corev1.ConditionTrue ||
+				c.Type != corev1.NodeReady && c.Status != corev1.ConditionFalse {
+				counters[metricsLabelMap[c.Type]] += 1
+			}
+		}
+	}
+
+	for _, v := range metricsLabelMap {
+		if counters[v] > 0 {
+			mon.emitGauge("cluster.nodes.condition", counters[v], map[string]string{"condition": string(v)})
```

Comment:
So this we will emit 1 metric / node / customer here ?

Reply:
no, 1 metric for each condition (with number of nodes expressing this condition) and ONLY IF this counter is more than 0 @julienstroheker

Reply:
1 metric for each non-zero condition per 1 cluster